app/models/recipe.rb

```diff
@@ -1,3 +1,24 @@
 class Recipe < ApplicationRecord
    validates_presence_of :name, :complexity, :genre
+
+   has_many :recipe_ingredients
+   has_many :ingredients, through: :recipe_ingredients
+
+   def all_ingredients
+      #there may be a better way to do this with AR but considering the circumstances this will do for now
+      list = []
+      self.ingredients.each do |ingredient|
+         list << ingredient.name
+      end
+
+      list.join(", ")
+   end
+   
+   def total_cost
+      cost = 0
+      self.ingredients.each do |ingredient|
+         cost += ingredient.cost
+      end
```

Comment:
Same situation here --- AR! 

`ingredients.sum(:cost)`

If you find yourself doing a `.each` in a model method, stop yourself and do some research to find out what AR method is available to make it waayyyyy easier